### ISSUE TITLE
Migrate to new extension registration

### ISSUE THREAD

**Issue by ingomueller-net** [jornane/mwSimpleSamlAuth#23]
The way extensions register themselves has changed. [This page](https://www.mediawiki.org/wiki/Manual:Extension_registration#Migration_for_extension_developers) explains what extension developers can do.

You seem like eager to work on your project at the moment, so I won't prepare a comment yet :)

**Comment by jornane** [comment]
In any case, I don't want the extension to *require* MediaWiki ≥ 1.25. I'll check if this can be done without duplicating all information about the extension in both the entry point and the JSON metadata file.

**Comment by ingomueller-net** [comment]
Good point. I am not sure how to remain backward compatible without duplicating the info though...

**Comment by jornane** [comment]
I don't think it's possible. So I think we should wait until doing this provides a real advantage, or 1.25 is the lowest supported version.

**Comment by ingomueller-net** [comment]
Your call. I just looked up the [lifecycle strategy](https://www.mediawiki.org/wiki/Version_lifecycle). At the latest when the LTS version 1.23 support ends, which happens in May 2017, I guess you can safely drop the old way.

**Comment by jamesmontalvo3** [comment]
Here's a method to handle both: https://github.com/enterprisemediawiki/MasonryMainPage/blob/master/MasonryMainPage.php
